src/scripts/components/VideoListItem.js

```diff
@@ -76,6 +107,14 @@ class VideoListItem extends Component<Props, void> {
   }
 
   handleClick () {
+    const form = document.getElementById('video-form')
+    if (form) {
+      const offsetTop = form.offsetTop - 20
+
+      if (offsetTop > window.innerHeight / 2) {
+        document.getElementById('root').scrollTo(0, offsetTop)
+      }
+    }
```

Comment:
This works, but there is a more react-way to get references to actual dom nodes 😄 (requires no hard-coded IDs).

Take a look at this line in VolumeBar.js. Here, we take advantage of styled-components's innerRef property. This provides a mechanism for us to access the ref property on the underlying react elements.

Long story short: ref/innerRef are properties that take a callback. This callback is invoked with an HTMLElement that you can then cache/store somewhere. VolumeBar.js is a good example.

Let me know if you have any questions about how this should work!

Reply:
I changed this code in https://github.com/Paratii-Video/paratii-portal/pull/324/commits/acceeb58b913e65813af3ed20a546c5fe1a4aff0. I'm trying to get the reference of `VideoForm` on `VideoListItem` from `VideoManagerContainer`.

When I [click on](https://github.com/Paratii-Video/paratii-portal/blob/acceeb58b913e65813af3ed20a546c5fe1a4aff0/src/scripts/components/VideoListItem.js#L109-L112) a `VideoListItem` the `this.props.videoFormRef` returns `null`. After somes clicks it get the `div.Card`.
if you click on a `VideoListItem` then click on `ADD ANOTHER VIDEO` then again clicks on `VideoListItem` and keep clicking on it(no change to other `VideoListItem`) the `this.props.videoFormRef` will always returns `null`:
![getref](https://user-images.githubusercontent.com/320157/37175903-a5aff680-22f9-11e8-941e-956c62a8aa00.gif)